tests/test_mortality.cpp

```diff
@@ -42,28 +42,57 @@ using std::endl;
 
 using namespace pops;
 
-int main(int argc, char* argv[])
+int test_mortality()
 {
     Raster<int> infected = {{5, 0}, {0, 0}};
-    Raster<int> mortality_tracker = {{{3, 0}, {0, 0}}, {{0, 0}, {0, 0}}};
-    Raster<int> mortality = {{0, 0}, {0, 0}};
+    Raster<int> total_hosts = {{10, 5}, {5, 3}};
+    std::vector<Raster<int>> mortality_tracker = {{{3, 0}, {0, 0}}, {{2, 0}, {0, 0}}};
+    Raster<int> died = {{0, 0}, {0, 0}};
+    Raster<int> expected_died = {{4, 0}, {0, 0}};
+    Raster<int> expected_infected = {{1, 0}, {0, 0}};
+    Raster<int> expected_total_hosts = {{6, 5}, {5, 3}};
     int ew_res = 30;
     int ns_res = 30;
     double mortality_rate = 0.50;
-    int current_year = 2018;
-    int first_mortality_year = 2018;
+    int mortality_time_lag = 0;
     std::vector<std::vector<int>> suitable_cells = {{0, 0}, {0, 1}, {1, 0}, {1, 1}};
-    Simulation<Raster<int>, Raster<double>> simulation(42, infected, ew_res, ns_res);
+    Simulation<Raster<int>, Raster<double>> simulation(
+        42, infected.rows(), infected.cols());
```

Comment:
Simulation constructor really takes resolution, not raster size. (Suggest a different naming if needed. These are short but perhaps impossible to parse out.)

The resolution is used for dispersal, so not in this test, so it could be even 0, anything as long as it is not misleading.

Reply:
The Simulation constructor takes rows and columns and config.ew_res and ns_res are set as part of the config setup.

Reply:
Wait, that's wrong. That is true for (some) kernels, but Simulation needs rows cols.